src/main/java/Functions.java

```diff
@@ -0,0 +1,142 @@
+import java.util.Scanner;
+public class Functions {
+
+    protected static final Scanner in = new Scanner(System.in);
+    private static final String SEPARATOR = "_________________________________________________________";
+    private Task[] taskList = new Task[100];
+    private int taskCounter = 0;
+
+    public Functions() {
+        printFunctions();
+    }
+
+    public void printFunctions(){
+        System.out.println("Currently, I am able to execute the following functions:");
+        System.out.println("1. Add tasks: I can add tasks to your task list.");
+        System.out.println("   - Type command: todo <task>");
+        System.out.println("   - Type command: deadline <task> /by <date>");
+        System.out.println("   - Type command: event <task> /from <day> <start time> /to <end time>");
+        System.out.println("2. Mark tasks as done:");
+        System.out.println("   - Type command: mark <task number>");
+        System.out.println("3. Mark tasks as not done:");
+        System.out.println("   - Type command: unmark <task number>");
+        System.out.println("4. List all tasks.");
+        System.out.println("   - Type command: list");
+        System.out.println("5. Be a parrot!");
+        System.out.println("   - Type command: echo");
+        System.out.println("6. Shut myself down. Use this if you no longer need me :(");
+        System.out.println("   - Type command: bye");
+        System.out.println(SEPARATOR);
+    }
+
+    public void taskmaster() {
+        while (true) {
+            String input = in.nextLine();
+            processCommand(input);
+        }
+    }
+
+    private void processCommand(String input) {
+        if (input.startsWith("mark ") || input.startsWith("unmark ")) {
+            handleMarking(input);
+        }
+        else {
+            switch (input) {
+                case "bye":
+                    Chatbot.printByeMessage();
+                    System.exit(0);
+
+                case "list":
+                    listTasks();
+                    break;
+
+                case "echo":
+                    echo();
+                    break;
+
+                default:
+                    addTask(input);
+            }
+        }
+    }
+
+    private void handleMarking(String input) {
+        try {
+            int taskNumIndex = Integer.parseInt(input.substring(input.indexOf(' ') + 1)) - 1;
+            if (taskNumIndex >= 0 && taskNumIndex < taskCounter) {
+                if (input.startsWith("mark ")) {
+                    taskList[taskNumIndex].setDone();
```

Comment:
Should the print messages be extracted out further to ensure that the same level of abstraction is used?

Reply:
this is a good idea actually.... idk why i so dumb